jenkins-integration/beaker/install/pe/60_classify_nodes.rb

```diff
@@ -0,0 +1,26 @@
+require 'puppet/gatling/config'
+require 'scooter'
+
+test_name "Classify PE agents via Node Classifier"
+
+def classify_pe_nodes(classifier, nodes)
+  production_id = classifier.get_node_group_id_by_name('Production environment')
+  nodes = group_by_environment(nodes)
+  nodes.each_pair do |env, node_configs|
+    node_configs.each do |config|
+      classifier.create_new_node_group_model(
```

Comment:
Fwiw, if you may run some or all of these tests repeatedly (i.e., run the suite, something fails, run parts of it again); or if there's a possibility this node group already exists for some reason, there's a `find_or_create_node_group_model` method available.  We've used it in a number of places in our scale testing setup code:

https://github.com/puppetlabs/pe_acceptance_tests/blob/4.0.x/setup/scale/install/20_classification/30_clamps.rb#L50-L52